### PR TITLE
Add stochastic muzero

### PR DESCRIPTION
**Pull request by ipsec**
## What?
Added minimal support to stochastic muzero by issue #77.

## Why?
To be able to train stochastic environments like 2048, poker, ...

## How?
Added Afterstate and Encoder models with configurations to be able to run it.
Only MLP models are created, not CNN.

## Fixes necessary
1. In the loss function from in the [last commit](https://github.com/EdanToledo/Stoix/commit/ba1f703ae55b8abfc35530bfd45975fdf774000b) the encoder must to receive an Observation like describe in the paper:
image

And here:

image

The pseudocode too in the line [931](https://gist.github.com/Mononofu/7548d8aa4bf94e12bc7eb7662fd60b56#file-stochastic_muzero-py-L931)

I don't know how to get the observation and to pass to the encoder in your code.

2. In the pseudocode the value target are calculated in every unroll step, lines [910](https://gist.github.com/Mononofu/7548d8aa4bf94e12bc7eb7662fd60b56#file-stochastic_muzero-py-L910) and [948](https://gist.github.com/Mononofu/7548d8aa4bf94e12bc7eb7662fd60b56#file-stochastic_muzero-py-L948)

I don't know how to this in your code.

### CODE REVIEW

**Comment by EdanToledo** [comment]
Amazing, thanks so much for doing this. I will review the code and do the necessary modifications as soon as i can. It will most likely have to be after next week.
